components/renku_data_services/session/k8s_client.py

```diff
@@ -198,3 +137,54 @@ async def get_server(self, name: str) -> BuildRun | None:
                 message=f"Expected to find 1 build run when getting run {name}, found {len(output)}."
             )
         return BuildRun.model_validate(output[0])
+
+
+class ShipwrightClient:
+    """The K8s client that combines a base client and a cache."""
+
+    def __init__(
+        self,
+        cache: ShipwrightCache | None,
+        base_client: ShipwrightClientBase,
+        # NOTE: If cache skipping is enabled then when the cache fails a large number of
+        # buildruns can overload the k8s API by submitting a lot of calls directly.
+        skip_cache_if_unavailable: bool = False,
+    ) -> None:
+        self.cache = cache
+        self.base_client = base_client
+        self.skip_cache_if_unavailable = skip_cache_if_unavailable
```

Comment:
issue: check and raise error if `skip_cache_if_unavailable` is `False` and `cache` is None. We should be skipping the cache only if the flag is set to true in no other scenarios.